arc/cli/commands/diff.py

```
@@ -44,30 +40,74 @@ async def _initialize_state():
+    
+    Returns list of scenario IDs that are in both runs.
+    """
+    scenarios1 = {s.get('scenario_id', s.get('id', f'scenario_{i}')) 
+                  for i, s in enumerate(run1.scenarios)}
+    scenarios2 = {s.get('scenario_id', s.get('id', f'scenario_{i}')) 
+                  for i, s in enumerate(run2.scenarios)}
+    
+    return list(scenarios1.intersection(scenarios2))
+
+
+def _filter_results_to_scenarios(results: List[Dict], scenario_ids: List[str]) -> List[Dict]:
```

Ruff (0.11.9)

Use `list` instead of `List` for type annotation

Replace with `list`

(UP006)